cf/regrid/regridoperator.py

```diff
@@ -487,54 +496,25 @@ def parameters(self):
             removed_at="5.0.0",
         )
 
-    def todense(self, order="C"):
-        """Return the weights in dense format.
-
-        .. versionadded:: 3.14.0
-
-        .. seealso:: `tosparse`
-
-        :Parameters:
-
-            order: `str`, optional
-                Specify the memory layout of the returned weights
-                matrix. ``'C'`` (the default) means C order
-                (row-major), and``'F'`` means Fortran order
-                (column-major).
-
-        :Returns:
-
-            `numpy.ndarray`
-                The 2-d dense weights matrix, an array with with shape
-                ``(J, I)``, where ``J`` is the number of destination
-                grid cells and ``I`` is the number of source grid
-                cells.
-
-        """
-        return self.tosparse().todense(order=order)
-
     def tosparse(self):
-        """Return the weights in sparse COOrdinate format.
-
-        See `scipy.sparse._arrays.coo_array` for sparse format
-        details.
-
-        .. versionadded:: 3.14.0
-
-        .. seealso:: `todense`
+        """Convert the weights to `scipy` sparse array format in-place.
 
         :Returns:
 
-            `scipy.sparse._arrays.coo_array`
-                The sparse array of weights.
+            `None`
 
         """
+        row = self.row
+        if row is None:
+            return
+
         from math import prod
 
-        from scipy.sparse import coo_array
+        from scipy.sparse import csr_array
```

Comment:
Regarding this change and your related comment:

> The change from coo_array to csr_array is because the latter is much more efficient at row slicing.

I notice the SciPy docs say that the CSR method has such an advantage but [also disadvantages of](https://docs.scipy.org/doc/scipy/reference/generated/scipy.sparse.csr_array.html):

> slow column slicing operations (consider CSC)
> changes to the sparsity structure are expensive (consider LIL or DOK)

but I believe those disadvantages aren't relevant in this use context, unless I am missing some case where the latter may be detrimental, and if so if you have considered those alternatives suggested for suitability?

Reply:
That's it - not going to change the sparsity, and will always do at least as many row slices as column slices. I'll make a note: d94e5cf5124e8b23cb517675b5b58bfcf6f2dce4